src/internal/pfssync/cache_client.go

```diff
@@ -0,0 +1,59 @@
+package pfssync
+
+import (
+	io "io"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pfsdb"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/renew"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+type CacheClient struct {
+	*client.APIClient
+	mu      sync.Mutex
+	cache   *simplelru.LRU
+	renewer *renew.StringSet
+}
+
+// TODO: Expose configuration for cache size?
+// TODO: Dedupe work?
+func NewCacheClient(pachClient *client.APIClient, renewer *renew.StringSet) *CacheClient {
+	cc := &CacheClient{
+		APIClient: pachClient,
+		renewer:   renewer,
+	}
+	cache, err := simplelru.NewLRU(100, cc.onEvicted)
```

Comment:
100*20MB = 2GB.  Is there a way to set this? Are we okay with saying the workers need this much RAM?

Reply:
This is caching the ID, not any chunks.